gems/smithy/lib/smithy/anvil/client/views/shapes.rb

```diff
@@ -0,0 +1,238 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Anvil
+    module Client
+      module Views
+        # @api private
+        class Shapes < View
+          def initialize(plan)
+            @plan = plan
+            @model = plan.model
+            @shapes, @operation_shapes, @service_shape = assemble_shapes
+            super()
+          end
+
+          attr_reader :shapes, :operation_shapes, :service_shape
+
+          def namespace
+            Tools::Namespace.namespace_from_gem_name(@plan.options[:gem_name])
+          end
+
+          def shapes_with_members
+            @shapes.reject { |s| s.members.empty? }
+          end
+
+          private
+
+          def assemble_shapes
```

Comment:
I would consider a private ShapesParser in anvil for hiding away most of this logic. You can define the "view classes" here though and populate them with data the shapes parser gives you. Also look at what vise gives you now.